src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -1528,12 +1541,12 @@ public override bool IsDBNull(int ordinal)
         /// <returns><b>true</b> if the specified column value is equivalent to <see cref="DBNull"/> otherwise <b>false</b>.</returns>
         public override async Task<bool> IsDBNullAsync(int ordinal, CancellationToken cancellationToken)
         {
+            CheckRowAndOrdinal(ordinal);
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Elsewhere you replaced `ThrowIfCancellationRequested()` with `Task.FromCanceled<T>(cancellationToken)`. Is there something different about this method/the caller context?

Reply:
Yes - this method is an async method, so an exception here would be automatically translated into a cancelled Task.

The confusion originally here is between async methods (i.e. methods with the async keyword), and methods which return a Task but without the async keyword - the latter need to manually return a cancelled task rather than throwing (that's basically what this PR is about).